crates/uv-pypi-types/src/requirement.rs

```diff
@@ -97,6 +95,47 @@ impl Requirement {
     }
 }
 
+impl std::hash::Hash for Requirement {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        let Self {
+            name,
+            extras,
+            marker,
+            source,
+            origin: _,
+        } = self;
+        name.hash(state);
+        extras.hash(state);
+        marker.hash(state);
+        source.hash(state);
+    }
+}
+
+impl PartialEq for Requirement {
+    fn eq(&self, other: &Self) -> bool {
+        let Self {
+            name,
+            extras,
+            marker,
+            source,
+            origin: _,
+        } = self;
+        let Self {
+            name: other_name,
+            extras: other_extras,
+            marker: other_marker,
+            source: other_source,
+            origin: _,
+        } = other;
+        name == other_name
+            && extras == other_extras
+            && marker == other_marker
+            && source == other_source
+    }
+}
```

Comment:
I think you need custom `Ord`/`PartialOrd` trait impls too, right? Otherwise, those will incorporate `origin` for comparisons and I think result in an inconsistency with your `Eq`/`PartialEq` trait impls.

Reply:
Classic :D

Reply:
Yeah this is truly classic. Thank you.